src/importlinter/domain/imports.py

```diff
@@ -83,6 +83,14 @@ def __hash__(self) -> int:
         return hash((str(self), self.line_contents))
 
 
+class ModuleExpression(ValueObject):
```

Comment:
This could do with a docstring, a bit like `ImportExpression`.

(I actually notice that the `ImportExpression` docstring is out of date, as it doesn't mention the `**` syntax. Maybe it's worth documenting that within the `ModuleExpression` docstring and then just referring to `ModuleExpression from the `ImportExpression` docstring, so we don't need to document it thoroughly in both places?)